doc/decisions/boolean.md

```diff
@@ -6,46 +6,39 @@ Inconsistent use of bool in various parts of Elektra.
 
 ## Constraints
 
+- needs to be string
+
 ## Assumptions
 
-- needs to be string
-- convenience plugins can convert anything to 0 or 1
 - type checker plugins can reject everything not 0 or 1
 
 ## Considered Alternatives
 
-- strictly only allow 0 and 1 (would move validation across the code)
 - only check presence or absence (no cascading override of already present key possible)
-- use as in CMake (would move convenience across the code)
+- use booleans as in CMake, which allows on/off, true/false, ... (would need convenience across the code)
 
 ## Decision
 
-Use, depending on what your default should be:
+Only the strings `0` and `1` are allowed in the `KeySet` for `type = boolean`.
+Everything else should lead to errors in checkers (in `kdbSet`).
 
-- 0 is false, everything else is true (default is true)
-- 1 is true, everything else is false (default is false)
+The spec/docu should mention that a `boolean` is used and may specify the default.
+The absence of a non-required key without a specified default should be interpreted as false.
 
-Example:
+Example for an implementation in C in an application:
 
 ```c
-if ( strcmp(keyString(k), "0")) {/*true*/} else {/*false*/}
 if (!strcmp(keyString(k), "1")) {/*true*/} else {/*false*/}
```

Comment:
```suggestion
if (k != NULL && strcmp(keyString(k), "1") == 0) {/*true*/} else {/*false*/}
```

I know the other code works as well, but we should use the most explicit and least confusing code variant in the documentation.